### PR TITLE
Lint client

### PR DESCRIPTION
**Pull request by ZeHiro**
[Review on Reviewable](https://reviewable.io/reviews/bnjbvr/kresus/321)

### CODE REVIEW

File: client/.eslintrc.yml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        # TODO : reset when using prop validation with propTypes
+        'react/forbid-prop-types':
+            0
+        # Enforce the explicit use of true when a prop is a boolean and set to true
+        'react/jsx-boolean-value':
+            0
+        # Location of the closing bracket for JSX objects
+        'react/jsx-closing-bracket-location':
+            - 2
+            - {selfClosing: 'after-props', nonEmpty: 'after-props'}
```
Comment (ZeHiro) [inline client/.eslintrc.yml]: selfClosing could be set to 'tag-aligned', to match current writing rules

**Comment by bnjbvr** [comment]
Thank you for the first set of rules, they look good! I've commented a few of them which look nice to have, if you can enable those and try to lint a small component, that would be awesome! Cheers

---

Reviewed 6 of 8 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 16 unresolved discussions.

---

*[client/.eslintrc.yml, line 29 \[r1\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7NXs4SMwHfGu3Vuxnu-r1-29:-K7YQ5oO-mkGH7r6vAzJ:-915358048) ([raw file](https://github.com/bnjbvr/kresus/blob/6f1b344084bc16aa02dcafa358cf9290ea8ebafc/client/.eslintrc.yml#L29)):*
Yes, agreed.

---

*[client/.eslintrc.yml, line 15 \[r2\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7YP1aNc6qYM2ox1Z3o:-K7YP1aNc6qYM2ox1Z3p:-369140612) ([raw file](https://github.com/bnjbvr/kresus/blob/9d0eb1349b72f5b9ecd4f48c22ec69f4328383a1/client/.eslintrc.yml#L15)):*
nit: typo in comment. For these comments, feel free to just copy/paste the textual description of the rule as described in https://www.npmjs.com/package/eslint-plugin-react#list-of-supported-rules

---

*[client/.eslintrc.yml, line 21 \[r2\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7YPIk3aOuG4vJWdXDm:-K7YPIk402fpgLclwOkp:-108021364) ([raw file](https://github.com/bnjbvr/kresus/blob/9d0eb1349b72f5b9ecd4f48c22ec69f4328383a1/client/.eslintrc.yml#L21)):*
If I understand correctly, as we're not using prop types yet, this can be enabled without provoking any validation errors? If so, can we just enable it already? (I'd forbid the 3 recommended values, "array", "object" and "any")

---

*[client/.eslintrc.yml, line 24 \[r2\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7YPjPQHOICCA_dxgsT:-K7YPjPQHOICCA_dxgsU:1808349005) ([raw file](https://github.com/bnjbvr/kresus/blob/9d0eb1349b72f5b9ecd4f48c22ec69f4328383a1/client/.eslintrc.yml#L24)):*
Let's enable this: explicit is better than implicit.

---

*[client/.eslintrc.yml, line 37 \[r2\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7YQimlgktxPE4FmeQz:-K7YQimlgktxPE4FmeR-:2136246374) ([raw file](https://github.com/bnjbvr/kresus/blob/9d0eb1349b72f5b9ecd4f48c22ec69f4328383a1/client/.eslintrc.yml#L37)):*
Looks nice. Let's enable it (without specifying parameters, so as to use the default ones).

---

*[client/.eslintrc.yml, line 42 \[r2\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7YQzL1pvzYDsSWy2Lo:-K7YQzL1pvzYDsSWy2Lp:1771384549) ([raw file](https://github.com/bnjbvr/kresus/blob/9d0eb1349b72f5b9ecd4f48c22ec69f4328383a1/client/.eslintrc.yml#L42)):*
nit in comment: ident => indent

---

*[client/.eslintrc.yml, line 49 \[r2\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7YRFYSRS1GAQOpWEq5:-K7YRFYSRS1GAQOpWEq6:-1584783498) ([raw file](https://github.com/bnjbvr/kresus/blob/9d0eb1349b72f5b9ecd4f48c22ec69f4328383a1/client/.eslintrc.yml#L49)):*
Please enable this one, it's going to cause performance faults if we miss keys at important locations.

---

*[client/.eslintrc.yml, line 52 \[r2\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7YRV1BSw3vfbDSb_5B:-K7YRV1BSw3vfbDSb_5C:-1066109114) ([raw file](https://github.com/bnjbvr/kresus/blob/9d0eb1349b72f5b9ecd4f48c22ec69f4328383a1/client/.eslintrc.yml#L52)):*
Let's enable this one with 1 prop per line.

---

*[client/.eslintrc.yml, line 55 \[r2\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7YR_9sHynBZDm_FYFE:-K7YR_9sHynBZDm_FYFF:195732409) ([raw file](https://github.com/bnjbvr/kresus/blob/9d0eb1349b72f5b9ecd4f48c22ec69f4328383a1/client/.eslintrc.yml#L55)):*
Let's enable it as well: we should not bind in handlers...

---

*[client/.eslintrc.yml, line 60 \[r2\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7YRhrKL2-A8bTBQKJy:-K7YRhrKL2-A8bTBQKJz:1246010376) ([raw file](https://github.com/bnjbvr/kresus/blob/9d0eb1349b72f5b9ecd4f48c22ec69f4328383a1/client/.eslintrc.yml#L60)):*
I agree: it's not needed for literals.

---

*[client/.eslintrc.yml, line 73 \[r2\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7YS03TAYhmk1qo-f14:-K7YS03UWiHCd6Lhd9_N:-1267864408) ([raw file](https://github.com/bnjbvr/kresus/blob/9d0eb1349b72f5b9ecd4f48c22ec69f4328383a1/client/.eslintrc.yml#L73)):*
No, keep it disabled; alphabetical order would show useful only if we had > 10 properties per object, which we don't.

---

*[client/.eslintrc.yml, line 85 \[r2\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7YSP6lRpKcLQJFEDdX:-K7YSP6m1D3l_WMuATGB:1096415757) ([raw file](https://github.com/bnjbvr/kresus/blob/9d0eb1349b72f5b9ecd4f48c22ec69f4328383a1/client/.eslintrc.yml#L85)):*
That prevents the use of props which can paste HTML, without checking it. Dangerously set HTML could be an issue if a user fills a form with HTML code: it would then be treated as HTML by the browser. On non-dangerous properties, actual tags are replaced by `<>`, which is safer. Let's enable it.

---

*[client/.eslintrc.yml, line 111 \[r2\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7YT7kxygjS2k8daBKG:-K7YT7kyjm_8RrcXsm7a:1278607513) ([raw file](https://github.com/bnjbvr/kresus/blob/9d0eb1349b72f5b9ecd4f48c22ec69f4328383a1/client/.eslintrc.yml#L111)):*
Can you add in the comment: "TODO: Reconsider this once we use Redux"?

---

*[client/.eslintrc.yml, line 114 \[r2\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7YTMK5FxzI2I9EEJyC:-K7YTMK6bPqTjE0SsKZF:-2054167946) ([raw file](https://github.com/bnjbvr/kresus/blob/9d0eb1349b72f5b9ecd4f48c22ec69f4328383a1/client/.eslintrc.yml#L114)):*
Please add a TODO, we'll do that later.

---

*[client/.eslintrc.yml, line 131 \[r2\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7YTgpQOnqWVx9n4a_v:-K7YTgpQOnqWVx9n4a_w:935631847) ([raw file](https://github.com/bnjbvr/kresus/blob/9d0eb1349b72f5b9ecd4f48c22ec69f4328383a1/client/.eslintrc.yml#L131)):*
Let's disable this, it seems a bit overzealous to consider file's extensions.

---

*[client/.eslintrc.yml, line 138 \[r2\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7YTveLkbbysHwf2wjr:-K7YTveLkbbysHwf2wjs:-1934277927) ([raw file](https://github.com/bnjbvr/kresus/blob/9d0eb1349b72f5b9ecd4f48c22ec69f4328383a1/client/.eslintrc.yml#L138)):*
I think it's too much complexity for a very low reward, let's just disable it.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-:-K7YUcjDH_NBcl96CD2F:-739614508)*

**Comment by ZeHiro** [comment]
3 components were linted with the rules. One error is not corrected as it has implications on the whole client. 

---

Review status: 6 of 8 files reviewed at latest revision, 1 unresolved discussion.

---

*[client/.eslintrc.yml, line 51 \[r3\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7aXljJd1JsYx7MeDuy:-K7aXljJd1JsYx7MeDuz:1173856190) ([raw file](https://github.com/bnjbvr/kresus/blob/d7a2659e35eadc3bcd5f0511ed9d8b39f58e3375/client/.eslintrc.yml#L51)):*
I have set this one to 3, as this rule also applies to raw html balises. 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-:-K7aYWQfQDK4FY0VRctT:179227756)*

**Comment by bnjbvr** [comment]
It's looking beautiful! Which error haven't you fixed yet, out of curiosity? I'd like to understand why it has these implications.

---

Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[client/.eslintrc.yml, line 21 \[r3\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7b0yrQWAEtGPuZNYOs:-K7b0yrQWAEtGPuZNYOt:1963740238) ([raw file](https://github.com/bnjbvr/kresus/blob/d7a2659e35eadc3bcd5f0511ed9d8b39f58e3375/client/.eslintrc.yml#L21)):*
Let's make it a strict error from the start: 2.

---

*[client/components/AddOperationModal.js, line 25 \[r3\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7b1d_Q9qoNKqvyrMML:-K7b1d_Rbh1thpCC0KTV:-202888763) ([raw file](https://github.com/bnjbvr/kresus/blob/d7a2659e35eadc3bcd5f0511ed9d8b39f58e3375/client/components/AddOperationModal.js#L25)):*
Please rename `onSubmit` as well, so that both hand sides have the same value.

---

*[client/components/AddOperationModal.js, line 27 \[r3\]](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-K7b1iOWHarTYZ7dwpH_:-K7b1iOWHarTYZ7dwpHa:-469765787) ([raw file](https://github.com/bnjbvr/kresus/blob/d7a2659e35eadc3bcd5f0511ed9d8b39f58e3375/client/components/AddOperationModal.js#L27)):*
ditto

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-:-K7b1s6Js0ErJreADoQG:8009087)*

**Comment by ZeHiro** [comment]
the error : A function with a name starting with an uppercase letter should only be used as a constructor  new-cap
It is triggered by Actions.CreateOperation. I should then update the store, lint it etc... I can do this, without proper lint.

---

Review status: 6 of 8 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-:-K7b45xf1rOoakWAFWrG:568235159)*

**Comment by bnjbvr** [comment]
OK, thank you! And thank you for the pull request, I am gonna merge this.

---

Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-:-K7bDjJFaYabdPLG9RRa:-617637275)*

**Comment by ZeHiro** [comment]
merge what ? This version, or the one with the fix of the last error? 

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bnjbvr/kresus/321#-:-K7bEWuLJT7T2J8hW5s3:-937043441)*

**Comment by bnjbvr** [comment]
This pull request only; let's blow up the entire client codebase another day :)

**Pull request merged**